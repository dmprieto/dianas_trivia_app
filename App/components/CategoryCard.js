import { Text, TouchableOpacity } from "react-native"

export const CategoryCard = ({ onPress, title }) => (
  <TouchableOpacity
    onPress={onPress}
    className="py-3 px-3 opacity-90 bg-purple-900 text-white text-lg font-semibold rounded-md mt-4"
  >
    <Text className="text-2xl text-purple-100">{title}</Text>
  </TouchableOpacity>
)
