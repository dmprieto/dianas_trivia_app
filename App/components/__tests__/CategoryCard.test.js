import renderer from "react-test-renderer"
import { CategoryCard } from "../CategoryCard"

describe("<CategoryCard />", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<CategoryCard />).toJSON()
    expect(tree.children.length).toBe(1)
  })

  it("renders correctly", () => {
    const tree = renderer.create(<CategoryCard />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
