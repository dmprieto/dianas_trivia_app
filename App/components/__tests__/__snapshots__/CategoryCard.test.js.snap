// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryCard /> renders correctly 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#581c87",
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "600",
        "lineHeight": 28,
        "marginTop": 16,
        "opacity": 0.9,
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 24,
            "lineHeight": 32,
          },
          Object {
            "color": "#f3e8ff",
          },
        ]
      }
    />
  </View>
</View>
`;
